File: 0409-longest-palindrome/0409-longest-palindrome.js
const longestPalindrome = (s) => {
  let map = {};
  for (const char of s) map[char] ? map[char]++ : map[char] = 1;
  let size = 0, one = 0;
  map = Object.values(map);
  for (const count of map) {
    if (count & 1) {
      one = 1;
      size += count - 1;
    } else {
      size += count;
    }
  }
  return size + one;
};

File: 0409-longest-palindrome/0409-longest-palindrome.py
class Solution:
  def longestPalindrome(self, s: str) -> int:
    d = defaultdict(int)
    for letter in s:
      d[letter] += 1
    ans = 0
    addition = 0
    for letter in d:
      if d[letter] & 1:
        addition = 1
        ans += d[letter] - 1
      else:
        ans += d[letter]
    return ans + addition